star-wars-films/src/PersonDetail/PersonDetail.js

```diff
@@ -0,0 +1,130 @@
+  //
+  // {finalFilmList.map((finalFilmList, idx) => {
+  //    return <span>{finalFilmList}, </span>
+  // })}
+
+  getHomePlanet(planet) {
+    const planetNames = ["Tatooine","Alderaan","Yavin IV","Hoth","Dagobah","Bespin","Endor","Naboo","Coruscant", "Kamino", "Geonosis", "Utapau", "Mustafar", "Kashyyyk", "Polis Massa", "Mygeeto", "Felucia", "Cato Neimoidia", "Saleucami", "Stewjon", "Eriadu", "Corellia", "Rodia", "Nal Hutta", "Dantooine", "Bestine IV", "Ord Mantell", "unknown", "Trandosha", "Socorro", "Mon Cala", "Chandrila", "Sullust", "Toydaria", "Malastare", "Dathomir", "Ryloth", "Aleen Minor", "Vulpter", "Troiken", "Tund", "Haruun Kal", "Cerea", "Glee Anselm", "Iridonia", "Tholoth", "Iktotch", "Quermia", "Dorin", "Champala", "Mirial", "Serenno", "Concord Dawn","Zolan","Ojom","Skako","Muunilinst","Shili","Kalee","Umbara","Jakku"]
+    const url = planet;
+    let planetNum = url.split('/').reverse()[1];
+    planetNum = planetNum - 1;
+    const finalPlanet = planetNames[planetNum];
+    return (
+      <span>
+        {finalPlanet}
+      </span>
+    );
+  };
+
+  render() {
+    const { name, height, mass, hair_color, skin_color, eye_color, birth_year, gender, homeworld } = this.state.person;
+    const films = this.state.person.films;
+    const peopleIdNum = this.props.match.params.peopleId;
+    const backgroundStyles = {
+        backgroundImage: 'url(\'/characters/character-' + peopleIdNum + '.jpg\')',
+    };
+    return (
+      <div className="page-content">
+        <section className="person-detail">
+          <div className="content">
+            <h1>{name}</h1>
+            <aside className="stats">
+              <h2>Vital Stats</h2>
+              <div className="stats--shape">
+                <div className="stat--chart">
+                  <dl><dt>Height:</dt><dd>{height} cm</dd></dl>
+                  <dl><dt>Mass:</dt><dd>{mass} kg</dd></dl>
+                  <dl><dt>Hair color:</dt><dd>{hair_color}</dd></dl>
+                  <dl><dt>Skin color:</dt><dd>{skin_color}</dd></dl>
+                </div>
+                <div className="stat--chart">
+                  <dl><dt>Eye color:</dt><dd>{eye_color}</dd></dl>
+                  <dl><dt>Birth year:</dt><dd>{birth_year}</dd></dl>
+                  <dl><dt>Gender:</dt><dd>{gender}</dd></dl>
+                  <dl><dt>Homeworld:</dt>{homeworld ? this.getHomePlanet(homeworld) : "unknown"}<dd></dd></dl>
+                </div>
+                <div className="stat--notes">
+                  <p> Appears in: <span className="emph">{films ? this.getFilmList(films) : "none" }</span></p>
+                </div>
+              </div>
+            </aside>
+          </div>
+
+          <div className="media" style={backgroundStyles}>
+            <div className="styleized-font">{name}</div>
+            <div className="styleized-font">{name}</div>
+          </div>
+        </section>
+      </div>
+    )
+  }
+}
+
+// these PropTypes do not appear to work:
+
```

Comment:
I could never gets these PropTypes to work. I could change them to any data type and I'd never get an error message.

Reply:
Isn't person in state rather than in this.props ?